RepoCleanup/Functions/MigrateXsdSchemasFunction.cs

```diff
@@ -0,0 +1,170 @@
+                }
+            }
+
+            logger.WriteLog();
+            await Task.Delay(1);
+
+            return;
+        }
+
+        private static async Task CreateFolderStructure(string repoFolder)
+        {
+            Directory.CreateDirectory($"{repoFolder}\\.altinnstudio");
+            Directory.CreateDirectory($"{repoFolder}\\altinn2");
+            Directory.CreateDirectory($"{repoFolder}\\shared");
+
+            await System.IO.File.WriteAllTextAsync(
+                $"{repoFolder}\\.altinnstudio\\settings.json",
+                "{\n  \"repotype\" : \"datamodels\"\n}");
+
+            await System.IO.File.WriteAllTextAsync(
+                $"{repoFolder}\\shared\\README.md",
+                "# Shared models");
+        }
+
+        private static string CollectMigrationWorkFolder()
+        {
+            Console.WriteLine("This operation requires a folder to which it can clone the datamodels repositories.");
+            string basePath = SharedFunctionSnippets.CollectInput("Provide folder name (should be empty): ");
+
+            if (!Directory.Exists(basePath))
+            {
+                Console.WriteLine("Can't find the specified folder!");
+                return CollectMigrationWorkFolder();
+            }
+
+            return basePath;
+        }
+
+        /// <summary>
+        /// List the GIT status of a repository
+        /// </summary>
+        /// <param name="org">Unique identifier of the organisation responsible for the app.</param>
+        /// <param name="repository">The name of the repository</param>
+        /// <returns>A list of changed files in the repository</returns>
+        public static List<string> Status(string localRepoPath)
+        {
+            List<string> repoContent = new List<string>();
+            using (var repo = new LibGit2Sharp.Repository(localRepoPath))
+            {
+                Commands.Stage(repo, "*");
+                RepositoryStatus status = repo.RetrieveStatus(new StatusOptions());
+                foreach (StatusEntry item in status)
+                {
+                    repoContent.Add(item.FilePath);
+                }
+            }
+
+            return repoContent;
+        }
+
+        private static async Task<List<string>> CollectOrgInfo()
+        {
+            List<string> orgs = new List<string>();
+
+            bool updateAllOrgs = CheckIfAllOrgs();
+
+            if (updateAllOrgs)
+            {
+                List<Organisation> organisations = await GiteaService.GetOrganisations();
+                orgs.AddRange(organisations.Select(o => o.Username));
+            }
+            else
+            {
+                Console.Write("\r\nProvide organisation name: ");
+
+                string name = Console.ReadLine();
+                orgs.Add(name.ToLower());
+            }
+
+            return orgs;
+        }
+
+        private static bool CheckIfAllOrgs()
```

Comment:
There is already a shared function snippet for this ShouldThisApplyToAllOrgs. It doesn't run in a loop if you don't answer y or n. If you need that then perhaps move this to the shared function snippets.

Reply:
Adopting shared function without changes.